src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -4228,6 +4228,34 @@ class spell_gen_clear_debuffs : public SpellScriptLoader
         }
 };
 
+// 169869 - Transformation Sickness
+class spell_gen_decimatus_transformation_sickness : public SpellScriptLoader
+{
+public:
+    spell_gen_decimatus_transformation_sickness() : SpellScriptLoader("spell_gen_decimatus_transformation_sickness") { }
+
+    class spell_gen_decimatus_transformation_sickness_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gen_decimatus_transformation_sickness_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* caster = GetCaster())
+                caster->SetHealth(caster->CountPctFromMaxHealth(10));
+        }
```

Comment:
void HandleScript(SpellEffIndex /*effIndex*/)
         {
            if (Unit* caster = GetCaster())
                uint32 hp = caster->CountPctFromMaxHealth(10);
                if (hp)
                    caster->SetHealth(hp);
         }
Maybe this better? Also add in database HealthRegen interrupt flag.